presidio-image-redactor/presidio_image_redactor/dicom_image_redactor_engine.py

```diff
@@ -0,0 +1,768 @@
+        new instance and manipulates it.
+
+        :param image: Loaded DICOM instance including pixel data and metadata.
+        :param fill: Fill setting to use for redaction box ("contrast" or "background").
+        :param padding_width: Padding width to use when running OCR.
+
+        :return: DICOM instance with redacted pixel data.
+        """
+        instance = deepcopy(image)
+
+        # Load image for processing
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            # Convert DICOM to PNG and add padding for OCR (during analysis)
+            is_greyscale = self._check_if_greyscale(instance)
+            image = self._rescale_dcm_pixel_array(instance, is_greyscale)
+            self._save_pixel_array_as_png(image, is_greyscale, "tmp_dcm", tmpdirname)
+
+            png_filepath = f"{tmpdirname}/tmp_dcm.png"
+            loaded_image = Image.open(png_filepath)
+            image = self._add_padding(loaded_image, is_greyscale, padding_width)
+
+        # Create custom recognizer using DICOM metadata
+        original_metadata, is_name, is_patient = self._get_text_metadata(instance)
+        phi_list = self._make_phi_list(original_metadata, is_name, is_patient)
+        deny_list_recognizer = PatternRecognizer(
+            supported_entity="PERSON", deny_list=phi_list
+        )
+        analyzer_results = self.image_analyzer_engine.analyze(
+            image, ad_hoc_recognizers=[deny_list_recognizer]
+        )
+
+        # Redact all bounding boxes from DICOM file
+        bboxes = self._format_bboxes(analyzer_results, padding_width)
+        redacted_image = self._add_redact_box(instance, bboxes, fill)
+
+        return redacted_image
+
+    def redact_from_file(
+        self,
+        input_dicom_path: str,
+        output_dir: str,
+        padding_width: int = 25,
+        fill: str = "contrast",
+    ) -> None:
+        """Redact method to redact the given image.
+
+        Please notice, this method duplicates the image, creates a
+        new instance and manipulate it.
+
+        :param input_dicom_path: String path to DICOM image(s).
+        :param output_dir: String path to parent output directory.
+        :param padding_width : Padding width to use when running OCR.
+        :param fill: Color setting to use for redaction box
+        ("contrast" or "background").
+        """
+        # Verify the given paths
+        self._validate_paths(input_dicom_path, output_dir)
+
+        # Create duplicate(s)
+        dst_path = self._copy_files_for_processing(input_dicom_path, output_dir)
+
+        # Process DICOM file(s)
+        if Path(dst_path).is_dir() is False:
+            output_location = self._redact_single_dicom_image(
+                dcm_path=dst_path,
+                fill=fill,
+                padding_width=padding_width,
+                overwrite=True,
+                dst_parent_dir=".",
+            )
+        else:
+            output_location = self._redact_multiple_dicom_images(
```

Comment:
From a user standpoint, I feel it would be convenient to have it built-in like we have here since DICOM files are often nested.

@omri374 - 2nd opinion?

Reply:
It seems that the user has the capability to define the iteration over files, and can call `redact_from_file` themselves. I don't have a strong oppinion here.

Reply:
@SharonHart , @omri374 - Would it make sense to split this into two functions then? `redact_from_file` and `redact_from_dir`? Perhaps that would bring a little more clarity.